ts/src/header-validator/source.test.ts

```diff
@@ -2544,6 +2559,223 @@ const testCases: TestCase[] = [
       },
     ],
   },
+
+  // Attribution Scope
+  {
+    name: 'attribution-scope-limit-negative',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": -1
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-zero',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 0
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-not-integer',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1.5
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be an integer',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-exceeds-max',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 4294967296
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'empty-attribution-scopes-with-limit',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 3,
+      "attribution_scopes": []
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [1, 3]',
+      },
+    ],
+  },
+  {
+    name: 'missing-attribution-scope-limit-attribution-scopes',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scopes": ["1", "2"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [0, 0]',
```

Comment:
We may want to add an explanation to this message, as it isn't obvious why the list has to be empty. In particular, we may want to hint that the `attribution_scope_limit` field should be added to avoid this.

Reply:
refactored the code a bit to make error message more specific